tasty-plutus/src/Test/Tasty/Plutus/Internal/Context.hs

```diff
@@ -106,13 +112,32 @@ data ExternalType
       Show
     )
 
+{- | Different types of value:
+
+  `TokensValue` is only used within `ContextBuilder ('ForMinting r)`
+  for representing classes, belonging to the tested MintingPolicy.
+
+  In all other cases `GeneralValue` is used.
+
+ @since 6.0
+-}
+data ValueType
+  = -- | @since 6.0
+    GeneralValue Value
+  | -- | @since 6.0
+    TokensValue TokenName Positive
+  deriving stock
+    ( -- | @since 6.0
+      Show
+    )
```

Comment:
No `Eq` instance?

Reply:
I was thinking of adding instances as needed. `ValueType` is internal type.
Do we need to add `Eq` to other internal types, such as `Input`, `Output` etc?

Reply:
Ah, OK. It's not really an issue, just a surprise. Especially if the type is internal, we can probably skip `Eq` instances.